cnd/src/swap_protocols/rfc003/events/mod.rs

```diff
@@ -10,36 +10,36 @@ use chrono::NaiveDateTime;
 use serde::{Deserialize, Serialize};
 
 #[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
-pub struct Funded<L: Ledger, A: Asset> {
-    pub transaction: L::Transaction,
+pub struct Funded<T, A> {
+    pub transaction: T,
     pub asset: A,
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
-pub struct Redeemed<L: Ledger> {
-    pub transaction: L::Transaction,
+pub struct Redeemed<T> {
+    pub transaction: T,
     pub secret: Secret,
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
-pub struct Deployed<L: Ledger> {
-    pub transaction: L::Transaction,
-    pub location: L::HtlcLocation,
+pub struct Deployed<T, H> {
+    pub transaction: T,
+    pub location: H,
 }
 
 #[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
-pub struct Refunded<L: Ledger> {
-    pub transaction: L::Transaction,
+pub struct Refunded<T> {
+    pub transaction: T,
 }
 
 #[async_trait::async_trait]
-pub trait HtlcFunded<L: Ledger, A: Asset>: Send + Sync + Sized + 'static {
+pub trait HtlcFunded<L: Ledger, A: Asset, T>: Send + Sync + Sized + 'static {
     async fn htlc_funded(
         &self,
         htlc_params: HtlcParams<L, A>,
-        htlc_deployment: &Deployed<L>,
+        htlc_deployment: &Deployed<T, L::HtlcLocation>,
```

Comment:
If you would have sticked to `L::Transaction` here, the change would be less invasive because now it bubbles up to all callers of `HtlcFunded`.

The ticket actually advised against that :see_no_evil: 
Not sure if you want to revert it or keep it :man_shrugging:

Reply:
This is the only trait where you did that. Are you gonna follow-up with the others or leave it like that?

Reply:
👍 Yes I am going to revert, I think I did something similar somewhere else and will revert too.

Reply:
I don't quite understand. You want to revert, yet you delegated to bors. Are you merging this and then sending a follow-up?

If you already decided to revert something, why merge it?

Reply:
Trust me.

Reply:
I do - I am just confused :D

Reply:
This is a huge change. I am trying to be kind to my colleagues are providing small chunks to review.
I have already fixed the problem locally and will open a follow-up pull request 😉